Ports content to a Docusarus v3 project (#267)
MozillaReality/hubs-docs: README.md

Reviewer: I think the Node/NPM version we recommend should be listed here in this section.  I know it was just added to `engines` in `package.json`, but I think it would be more prominent/helpful; if it was here in these docs as well.  E.g. you can just follow the docs to get started.

Note 1: I don't think we use Typescript, but if we doe, then it should probably be added hear as well.

Note 2: I kinda expected this to use Node v22 (which does seem to work for me), so I'm curious as to why Node v20 was picked over v22.

Reviewer 2: I'm transposed many of the concerns in @Exairnous' comment to be Documentation Session agenda items for April 22, 2026. I predict it will be possible to cover them there. However, those Session notes will only be about our discussion. I suggest that any future exploration of these comments (i.e. Why did they decide this?) should refer to **both** documentation sources.

In the Documentation Session agenda, I've separated the comments into 4 categories
1. Typos/coding changes
2. V3 port breaks, so proposed fixes
3. Pre-existing (V1) breaks that are continuing concerns (V3 may have exacerbated but did not _essentially cause_ these)
4. Pre-existing (V1) breaks that are continuing but we are not concerned

Reviewer 2: @Exairnous  On this

> Is it just me or is the text size bigger on v3? @hobbs-Hobbler is this good for accessibility or would we prefer it smaller so more text fits on the page at once? Also not sure if this is something that can be easily changed @DougReeder ?

I have found 2 general pieces of advice:
1. Minimum regular font size should be 12pt (16px) and large font size should be 18pt (24px) (but there are more involved tests that we don't need to do now that can be done to test font size in an assumed 320px-wide screen).

So what does the code say now? Does it use terms like "regular" and "large" or does it use an actual numbered point size?  Before answering, see the next item.

2. Using relative font sizes is preferred over an actual size number, because it is more responsive to different devices. Hence the use of terms like 'regular' and 'large'.  Do we use those terms?  Could also be terms like 'strong' or 'em'

[How to Pick the Perfect Font Size: A Guide to WCAG Accessibility ](https://www.a11y-collective.com/blog/wcag-minimum-font-size/)

My personal opinion is that I like the larger font size.

Reviewer: @DougReeder Bumping about typescript.

The documentation is under active development. If there are any changes or updates you recommend, feel free to submit a pull request or let us know in our [Discord Server](https://discord.gg/wHmY4nd).

This website was created with [Docusaurus v3](https://docusaurus.io/).

## What's In This Document

* [Get Started in 5 Minutes](#get-started-in-5-minutes)
* [Directory Structure](#directory-structure)
* [Editing Content](#editing-content)
* [Adding Content](#adding-content)
* [Full Documentation](#full-documentation)
* [Contributing to Hubs Docs](#contributing-to-hubs-docs)

> * [Docusaurus Header](#ddocusaurus-header)

Reviewer: Should this be the following?
```suggestion
* [Docusaurus Header](#docusaurus-header)
```

Edit me...
```

> See the [MDX syntax guide](https://github.com/micromark/mdx-state-machine#72-deviations-from-markdown) to understand how MDX differs from [standard Markdown](https://commonmark.org/help/).

Reviewer: The MDX syntax guide you link to is on an archived GitHub repository, plus it isn't really a syntax guide (IMO), so how about this?  CC @hobbs-Hobbler 

```suggestion
See the [Markdown syntax guide](https://commonmark.org/help/) for the basics, then check [the differences MDX introduces](https://docusaurus.io/docs/next/markdown-features/react#markdown-and-jsx-interoperability) to avoid running into any gotchas.
```

Reviewer: > The MDX syntax guide you link to is on an archived GitHub repository, plus it isn't really a syntax guide (IMO), so how about this? CC [@hobbs-Hobbler](https://github.com/hobbs-Hobbler)
> 
> Suggested change
> See the [MDX syntax guide](https://github.com/micromark/mdx-state-machine#72-deviations-from-markdown) to understand how MDX differs from [standard Markdown](https://commonmark.org/help/).
> See the [Markdown syntax guide](https://commonmark.org/help/) for the basics, then check [the differences MDX introduces](https://docusaurus.io/docs/next/markdown-features/react#markdown-and-jsx-interoperability) to avoid running into any gotchas.

Bump.  This still needs to be addressed.

The main things to watch out for are:
1. all tags must be closed, even void tags, for example: `<img src="foo.jpeg" alt="bar"/>`
2. left angle brackets that don't start a tag must be escaped with a backslash

> 3. left curly braces must be escaped with a backslash

Reviewer: Given that some of your fixup commits in this escape the right angle brackets/curly braces, maybe we should recommend they escape both when the raw character is wanted?  CC @hobbs-Hobbler

Reviewer: > Given that some of your fixup commits in this escape the right angle brackets/curly braces, maybe we should recommend they escape both when the raw character is wanted? CC [@hobbs-Hobbler](https://github.com/hobbs-Hobbler)

Bump. This still needs to be addressed.

Author: The right angle brackets and curly braces don't need to be escaped.  My early work did, as I didn't fully understand the rules when starting out.  I have no objection to people tweaking this, but I don't think it's worth our time to discuss this.

Reviewer: Okay.  Let's leave this text as is and see what people gravitate towards.

Edit me...
```

> For more information about blog posts, see [Adding a Blog in the Docusarus docs](https://docusaurus.io/docs/en/adding-blog).

Reviewer: This still links to the old v1 documentation, so it should probably be this instead:

```suggestion
For more information about blog posts, see [Adding a Blog in the Docusarus docs](https://docusaurus.io/docs/blog).
```

Reviewer: > This still links to the old v1 documentation, so it should probably be this instead:
> 
> Suggested change
> For more information about blog posts, see [Adding a Blog in the Docusarus docs](https://docusaurus.io/docs/en/adding-blog).
> For more information about blog posts, see [Adding a Blog in the Docusarus docs](https://docusaurus.io/docs/blog).

Bump. This still needs to be addressed.

My new content here..
```

1. Refer to that doc's ID in an existing sidebar in `website/sidebars.js`:

```javascript
const sidebars = {
  docs: [
    {
      type: 'category',
      label: 'Introduction',
      items: [
        'welcome',
        'intro-hubs',

>         'intro-spoke',

Reviewer: The line after `'intro-spoke'` should be `"newly-created-doc" // new doc here` so it's clear where people should add a new doc.

Reviewer: > The line after `'intro-spoke'` should be `"newly-created-doc" // new doc here` so it's clear where people should add a new doc.

Bump.  This still needs to be addressed.

  ...
}
```

> For more information about IDs and navigation, see [Navigation and Sidebars in the Docusaurus docs](https://docusaurus.io/docs/en/navigation).

Reviewer: This links back to the old v1 docs, so I'm guessing this link would work better here (and be more in line with the original text).  CC @hobbs-Hobbler 
```suggestion
For more information about adding new docs, see [Create a doc](https://docusaurus.io/docs/create-doc).
```

Reviewer 2: I had to scroll up to see the section header, **Adding a new docs page to an existing sidebar**, to figure out if this 'more information' was about

A. adding new docs (as in a whole new implementation of documentation)

or

B. adding a new doc (singular, adding a new document).

It's B. 
While @Exairnous suggested link appears to be about only adding a new doc, it does address the navigation issue as well. So, I'm OK with it.

### Step 3: Make Changes

Make changes to `.mdx` files in the `docs/` directory. Remember to:

- Add a Docusaurus header at the top
- Write atomic commit messages (see [Commit Message Guidelines](https://github.com/Hubs-Foundation/policies-procedures-guidelines-public/blob/main/commit-message-guidelines.md))

> - Follow the [MDX syntax guide](https://github.com/micromark/mdx-state-machine#72-deviations-from-markdown)

Reviewer: The MDX syntax guide you link to is on an archived GitHub repository, plus it isn't really a syntax guide (IMO), so how about this?  CC @hobbs-Hobbler 

```suggestion
Follow the [Markdown syntax guide](https://commonmark.org/help/) for the basics, then check [the differences MDX introduces](https://docusaurus.io/docs/next/markdown-features/react#markdown-and-jsx-interoperability) for additional functionality and to avoid running into any gotchas.
```

Reviewer 2: I have no strong opinion here.

### Step 3: Make Changes

Make changes to `.mdx` files in the `docs/` directory. Remember to:

- Add a Docusaurus header at the top
- Write atomic commit messages (see [Commit Message Guidelines](https://github.com/Hubs-Foundation/policies-procedures-guidelines-public/blob/main/commit-message-guidelines.md))

> Follow the [Markdown syntax guide](https://commonmark.org/help/) for the basics, then check [the differences MDX introduces](https://docusaurus.io/docs/next/markdown-features/react#markdown-and-jsx-interoperability) for additional functionality and to avoid running into any gotchas.

Reviewer: My previous suggestion here accidentally removed the bullet from this line.

This is my new suggestion:

```suggestion
- Follow the [Markdown syntax guide](https://commonmark.org/help/) for the basics, then check [the differences MDX introduces](https://docusaurus.io/docs/next/markdown-features/react#markdown-and-jsx-interoperability) for additional functionality and to avoid running into any gotchas.
```
